utmosv2/utils/_task_dependents/initializers.py

```diff
@@ -93,8 +95,14 @@ def get_model(cfg, device: torch.device) -> nn.Module:
                 Path("models")
                 / cfg.weight
                 / f"fold{cfg.now_fold}_s{cfg.split.seed}_best_model.pth"
-            ).as_posix()
-        model.load_state_dict(torch.load(weight_path))
+            )
+            if weight_path.exists():
+                model.load_state_dict(torch.load(weight_path.as_posix()))
+            else:
+                weight_path = f"https://huggingface.co/spaces/sarulab-speech/UTMOSv2/resolve/main/models/{cfg.weight}/fold{cfg.now_fold}_s{cfg.split.seed}_best_model.pth"
+                with smart_open.open(weight_path, "rb") as f:
+                    buf = io.BytesIO(f.read())
+                    model.load_state_dict(torch.load(buf))
```

Comment:
When specifying a `.pth` file directly, it wouldn’t be loaded correctly.

Also, consider using `download_pretrained_weights_from_hf`. Since downloading every time is time-consuming, caching it would be beneficial. It also prevents duplicate management.

```suggestion
            if not weight_path.exists():
                weight_path_cache = _UTMOSV2_CHACHE / weight_path
                if not weight_path_cache.exists():
                    download_pretrained_weights_from_hf(cfg.weight, cfg.now_fold)
                weight_path = weight_path_cache
        model.load_state_dict(torch.load(weight_path))
```

> [!NOTE]
> The imports must be fixed according to this change.

Reply:
Changed according to your suggestion. Also, I moved _constants.py and _download.py into utils/. This is to avoid circular imports caused by importing _core module.

Reply:
Thank you for your update!
Moving these modules to `utils/` is great since the changes here would otherwise cause circular dependencies.